PGwire query cancellation is now supported

rmloveland opened the pull request:
Fixes DOC-2590

Summary of changes:

- Update the 'Postgres compatibility' page to remove pgwire query
  cancellation from the list of unsupported features

- Add `sql.pgwire_cancel.*` metrics to 'Custom Chart Debug Page'

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/13832)
<!-- Reviewable:end -->

rafiss reviewed (commented):
lgtm! do you think this would be worth a callout on the CANCEL docs (https://www.cockroachlabs.com/docs/stable/cancel-query.html) that explains why it's different? for reference, here's what postgres says about it: https://www.postgresql.org/docs/14/protocol-flow.html#id-1.10.5.7.9

additionally, maybe it would be important to say that we will reject pgwire cancellation if there are too many unsuccessful attempts

rmloveland commented:
Thanks for the review Rafi! I made some updates based on your feedback

> do you think this would be worth a callout on the CANCEL docs (https://www.cockroachlabs.com/docs/stable/cancel-query.html) that explains why it's different? for reference, here's what postgres says about it: https://www.postgresql.org/docs/14/protocol-flow.html#id-1.10.5.7.9

good point, I went ahead and added some info on pgwire cancellation vs CANCEL QUERY to those docs

> additionally, maybe it would be important to say that we will reject pgwire cancellation if there are too many unsuccessful attempts

also added to the same updated section on CANCEL QUERY docs

PTAL and let me know what you think!

rmloveland commented:
yayyyy thanks Rafi!  sending on for docs review now

ericharmeling reviewed (approved):
one nit

otherwise, LGTM!

rmloveland commented on v22.1/cancel-query.md:
```diff
@@ -12,6 +12,10 @@ The `CANCEL QUERY` [statement](sql-statements.html) cancels a running SQL query.
 - Schema changes are treated differently than other SQL queries. You can use <a href="show-jobs.html"><code>SHOW JOBS</code></a> to monitor the progress of schema changes and <a href="cancel-job.html"><code>CANCEL JOB</code></a> to cancel schema changes that are taking longer than expected.
 - In rare cases where a query is close to completion when a cancellation request is issued, the query may run to completion.
+- In addition to the `CANCEL QUERY` statement, CockroachDB also supports query cancellation by [client drivers and ORMs](install-client-drivers.html) using the Postgres wire protocol (pgwire). This allows CockroachDB to stop executing queries that your application is no longer waiting for, thereby reducing load on the cluster. pgwire query cancellation differs from the `CANCEL QUERY` statement in ...
+  - It is how most client drivers and ORMS implement query cancellation. For example, it is [used by PGJDBC](https://github.com/pgjdbc/pgjdbc/blob/3a54d28e0b416a84353d85e73a23180a6719435e/pgjdbc/src/main/java/org/postgresql/core/QueryExecutorBase.java#L171) to implement the [`setQueryTimeout` method](https://jdbc.postgresql.org/documentation/publicapi/org/postgresql/jdbc/PgStatement.html#setQuery ...
```
I have no idea why, but when I delete the `-int-` text from the  link it no longer works (tested in Safari, Chrome, FF)

ericharmeling replied:
Hmmm... okay! LGTM :)

rmloveland commented:
TFTR Eric!
